### ISSUE TITLE
Add option diff that when used along with output-check shows differences

### ISSUE THREAD

**Issue by anilkumarmyla** [terraform-docs/terraform-docs#688]
### What problem are you facing?
Often times it happens that local and automation runs of terraform-docs produce slightly different versions of markdown due to various reasons and if the automation checks for up-to-date versions of markdown files with --output-check, it only shows whether the file is up-to-date or not without showing the actual diff between the produced file and what is in scm. Debugging this becomes harder as you need to replicate the exact environment of the automation and run it to see the difference. 

### How could terraform-docs help solve your problem?
If there was a diff option ex: `terraform-docs --output-check -diff .` that actually showed you the differences it would become much easier to pin-point why there is a difference in generated markdown v/s what is committed in scm

**Comment by jason-johnson** [comment]
Why can't your SCM do this?  There are plenty of tools for doing diffs, I don't see the added complexity to this tool to support this being worthwhile and it goes against the Unix philosophy of having specific tools for specific jobs.

You could have a step in your automation to copy the output file, then run terraform-docs and then run a diff on the two files.

**Comment by anilkumarmyla** [comment]
Yep, SCM is the workaround and we already do it. We can do this to figure out the differences, but is a bunch of commands. If it was a feature in the tool, its easy great, not a must
```console
terraform-docs .
git diff
git checkout -- .
```

**Comment by anilkumarmyla** [comment]
Also for context, terraform natively supports the diff option for `fmt` command
```
terraform fmt -check -diff
```
